src/store.cpp

```diff
@@ -0,0 +1,222 @@
+#include "store.h"
+#include <vector>
+
+using json = nlohmann::json;
+int store::setupSerial() {
+	
+    QSerialPort* serial= new QSerialPort();
+    serial->setPortName(this->dev);
+    serial->setBaudRate(QSerialPort::Baud115200);
+    serial->setDataBits(QSerialPort::Data8);
+    serial->setStopBits(QSerialPort::OneStop);
+    serial->setParity(QSerialPort::NoParity);
+    serial->setFlowControl(QSerialPort::NoFlowControl);
+    if (!serial->open(QIODevice::ReadWrite)) {
+        qDebug() << "Can't open " << this->dev << ", error code" << serial->error();
+        return 1;
+    }
+
+    this->port = serial;
+    connect(this->port, &QSerialPort::readyRead, this, &store::handleReadyRead);
+    connect(this->port, &QSerialPort::errorOccurred, this, &store::handleError);
+	
+
+	return 0;	
+}
+
+store::store( QString dev, QObject *parent  ): QObject(parent){
+	//if Qstring dev is empty, use default device
+	if(dev.isEmpty()){
+		this->dev = DEFAULT_DEVICE;
+	}else{
+		this->dev = dev;
+	}
+
+	setupSerial();
+	
+}
+void store::forceRead(qint64 len){
+	while (port->bytesAvailable() < len) {
+		if (!port->waitForReadyRead(1000)) {
+			qDebug() << QObject::tr("Wait read error %1")
+						.arg(port->errorString());
+			return;
+		}
+		
+	}
+    port->read(bufferMessage.data(), len);
+}
+//destructor for store 
+
+void store::handleReadyRead(){
+
+    bufferMessage=port->readAll();
+    serialLog.append(bufferMessage);
+	//can be optimized using pointers or even a variable as a "bookmark" wether a int or pointer 
+	lastMessage.append(bufferMessage);
+
+	//regex BSON_WARNING value in bufferMessage if its found set markerBSON_WARNING to the first character after BSON_WARNING
+    QString* temp = new QString(lastMessage);
+	if(temp->contains(BSON_WARNING)){
+		bsonMining();
+    }
+	
+
+}
+void store::bsonMining(){
+	//lastMessage starts at lastMessage size - ((lenght of buffer - index of BSON_WARNING) + length of BSON_WARNING)
+	const int bson_len = strlen(BSON_WARNING);
+	//im assuming a minimum of 9 due to warning=4, document size = 4, EOF =1, sum of them all is 9
+    if(lastMessage.size()<=9 || lastMessage.indexOf(BSON_WARNING)==-1){
```

Comment:
make a constant at the beginning to change the minimum (in this case 9).

Reply:
Will do